### PR TITLE
Added Curriculum class and its functionalities

### PR DESCRIPTION
**Pull request by Opeoluwa112**
#Story  
Added class for curriculum

#Description
The curriculum class contains functions for adding, removing and modifying a curriculum. It takes in attributes for id, title, color, description, list of materials and steps to be taken

### CODE REVIEW

**Comment by Opeoluwa112** [comment]
At some point in time in the system, don't we need to store the curriculum added, in a curriculum  array before a project is created ?

File: src/client/src/models/Curriculum.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    removeCurriculum(curriculum) {
+        for (let i = 0; i < this.curriculums.length; i++) {
+            if (curriculum.id == this.curriculums[i].id) {
+                this.curriculums.splice(i, 1);
+                break;
+            }
+        }
+    }
+
+    modifyCurriculum(title, description, color, materials, steps) {
```
Comment (karim2506) [inline src/client/src/models/Curriculum.js]: As the code will get bigger, I think we should be consistent from the beginning. For the Employee class, we decided to call it the "edit()" function, but here it is modifyCurriculum(). Same goes for addCurriculum() (or simply add() ) and removeCurriculum() (or simply remove() )?

@merekorlowski  any thoughts?

File: src/client/src/models/Curriculum.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        this.curriculums = [];
+        this.materials = [];
+        this.steps = [];
+    }
+
+    addCurriculum(curriculum) {
+        this.curriculums.push(curriculum);
+    }
+
+    removeCurriculum(curriculum) {
```
Comment (karim2506) [inline src/client/src/models/Curriculum.js]: Attention to be brought here for naming convention.

File: src/client/src/models/Curriculum.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        this.id = id;
+        this.title = title;
+        this.description = description;
+        this.colour = colour;
+        this.curriculums = [];
+        this.materials = [];
+        this.steps = [];
+    }
+
+    addCurriculum(curriculum) {
```
Comment (karim2506) [inline src/client/src/models/Curriculum.js]: Attention to be brought here for naming convention.

File: src/client/src/models/Curriculum.js (hunk 1 above)
Comment (merekorlowski) [reply src/client/src/models/Curriculum.js]: Yes I agree. However in this case, we will be removing add and remove from this class, as there will not be an array of curriculums in the Curriculum object.

**Pull request closed without merging**